Tools/TestWebKitAPI/Tests/WebKitCocoa/RestoreLocalStorage.mm

```diff
@@ -60,22 +62,58 @@ static void testFetchLocalStorage(RetainPtr<WKWebsiteDataStore> websiteDataStore
     RetainPtr set = [NSSet setWithObject:WKWebsiteDataTypeLocalStorage];
 
     __block bool doneFetching = false;
+    __block RetainPtr<NSData> localStorageData;
     [[[webView configuration] websiteDataStore] _fetchDataOfTypes:set.get() completionHandler:^(NSData *data) {
         EXPECT_NOT_NULL(data);
         EXPECT_GT([data length], 0u);
+        localStorageData = data;
         doneFetching = true;
     }];
     TestWebKitAPI::Util::run(&doneFetching);
+
+    // Restore Local Storage.
+
+    [websiteDataStore _terminateNetworkProcess];
+    [webView _killWebContentProcessAndResetState];
+
+    RetainPtr<WKWebsiteDataStore> newWebsiteDataStore;
+    if ([websiteDataStore isPersistent]) {
+        [WKWebsiteDataStore _deleteDefaultDataStoreForTesting];
+        newWebsiteDataStore = adoptNS([WKWebsiteDataStore defaultDataStore]);
+    } else
+        newWebsiteDataStore = adoptNS([WKWebsiteDataStore nonPersistentDataStore]);
+
+    RetainPtr newViewConfiguration = adoptNS([WKWebViewConfiguration new]);
+    [newViewConfiguration setWebsiteDataStore:newWebsiteDataStore.get()];
+    RetainPtr newWebView = adoptNS([[WKWebView alloc] initWithFrame:CGRectMake(0, 0, 100, 100) configuration:newViewConfiguration.get()]);
+
+    [[[newWebView configuration] websiteDataStore] _restoreDataOfTypes:set.get() data:localStorageData.get()];
+
+    WTF::sleep(1_s);
```

Comment:
I'm certain that this is wrong. Since `restoreDataOfTypes` will use async IPC but doesn't need a completion handler, how else can I wait for call the finish? It seems wrong to add a completion handler only for testing purposes.

Reply:
I personally think the WKDataStore SPI itself _should_ have a completion handler, because it is asynchronous in actuality.

Reply:
That's a good point. I can't think of any other way the client can know when the function is finished. I've added a completion handler.